### ISSUE TITLE
aws sdk using instance profile iam instead of IAM Roles for Service Accounts on eks 1.15

### ISSUE THREAD

**Issue by aavileli** [aws/aws-sdk-java#2283]
Please change the order of Default Credential Provider Chain 
https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.html

We are running springboot containers with springboot-cloud-aws-context dependency on eks worker nodes which have a default instance profile for the ec2 node. We are using IAM Roles for Service Accounts (IRSA) for the pods and the aws java sdk v1 always uses the instance profile instead of the Web Identity Token

As the instance does not have the required permission we get access denied.

How can I force the sdk to use Web Identity Token. Since EKS (via a webhook) automatically injected the environment variables AWS_IAM_ROLE_ARN and AWS_WEB_IDENTITY_TOKEN_FILE as well as the aws-iam-token why is the aws sdk not using it.

we are using https://github.com/spring-cloud/spring-cloud-aws package and have updated the core to use the latest aws sdk 1.11.758

* AWS Java SDK version used: 1.11.758
* JDK version used: amazon corretto 1.8
* Operating System and version: linux

**Comment by aavileli** [comment]
found a way to use AWS_WEB_IDENTITY_TOKEN_FILE. If any one reads this information the min aws sdk version for eks 704 and resquires aws-sdk-sts also be included for assume WEB_IDENTITY to work.

**Comment by Warfront1** [comment]
Thank you @aavileli I have spent far to long debugging this. Had you not mentioned this I may not have tried it.
For anyone else running into the same issue as aavileli mentioned added the aws sdk dependencies specifically for eks and sts.

I was also working with `org.springframework.cloud:spring-cloud-dependencies` `Hoxton.SR11`. By the way try to update if you are not at SR11 yet, as older version may not be up to date enough (AWS says it requires [SDK version 1.11.623+](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/)).
On Hoxton SR11 it pulls in AWS deps from AWS sdk version `1.11.792`, so I just added the following to my build.gradle .

```
    implementation group: 'com.amazonaws', name: 'aws-java-sdk-sts', version: '1.11.792'
    implementation group: 'com.amazonaws', name: 'aws-java-sdk-eks', version: '1.11.792'
```
After making that change everything worked perfectly as described in the AWS [documentation](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/):
